datalad/core/local/create.py

```diff
@@ -423,10 +423,10 @@ def __call__(
         # the next only makes sense if we saved the created dataset,
         # otherwise we have no committed state to be registered
         # in the parent
-        if isinstance(dataset, Dataset) and dataset.path != tbds.path:
+        if isinstance(refds, Dataset) and refds.path != tbds.path:
             # we created a dataset in another dataset
             # -> make submodule
-            for r in dataset.save(
+            for r in refds.save(
```

Comment:
oh, I blindly thought that the issue is more global than just `create` , but upon check -- 
<details><summary>`install` (I guess it is not revolutionary yet?) works as expected indeed</summary>

```shell
$> p=/tmp/testds; sudo rm -rf $p; datalad create $p; cd $p; datalad create -d . subm1 && git status && datalad install -d . /// && git status
[sudo] password for yoh:   
[INFO   ] Creating a new annex repo at /tmp/testds 
create(ok): /tmp/testds (dataset)                                                                                 
[INFO   ] Creating a new annex repo at /tmp/testds/subm1 
create(ok): subm1 (dataset)                                                                                       
On branch master
Untracked files:
  (use "git add <file>..." to include in what will be committed)

	subm1/

nothing added to commit but untracked files present (use "git add" to track)
[INFO   ] Cloning http://datasets.datalad.org/ [1 other candidates] into '/tmp/testds/datasets.datalad.org' 
install(ok): datasets.datalad.org (dataset)                                                                       
action summary:
  add (ok: 2)
  install (ok: 1)
  save (ok: 1)
On branch master
Untracked files:
  (use "git add <file>..." to include in what will be committed)

	subm1/

nothing added to commit but untracked files present (use "git add" to track)
```
</details>